central/node/datastore/store/postgres/store_test.go

```diff
@@ -96,6 +97,29 @@ func (s *NodesStoreSuite) TestStore() {
 	s.Nil(foundNode)
 }
 
+func (s *NodesStoreSuite) TestWalkByQuery() {
+	store := New(s.pool, false, concurrency.NewKeyFence())
+
+	node := &storage.Node{}
+	s.NoError(testutils.FullInit(node, testutils.UniqueInitializer(), testutils.JSONFieldsFilter))
+
+	node2 := node.CloneVT()
+	node2.Id = uuid.NewDummy().String()
+
+	s.NoError(store.Upsert(s.ctx, node))
+	s.NoError(store.Upsert(s.ctx, node2))
+
+	walkFn := func(obj *storage.Node) error {
+		if obj.GetId() != node.GetId() {
+			return fmt.Errorf("expected node1 but got %s", obj.GetId())
+		}
+		return nil
+	}
+
+	q := search.NewQueryBuilder().AddExactMatches(search.NodeID, node.GetId()).ProtoQuery()
+	s.NoError(store.WalkByQuery(s.ctx, q, walkFn))
+}
+
```

Comment:
nit: this seems unrelated to the PR's intent/changes? Or am I missing a connection between images and nodes?

Reply:
Yeah it is unrelated. I had added it to debug those context cancelled errors, but then decided to keep it as a test for WalkByQuery wasn't there for nodes.